pkg/image/manifest.go

```diff
@@ -199,62 +296,50 @@ func GetDefaultImageRegistries(version string) (*RegistryList, error) {
 	switch v.Segments()[0] {
 	case 1:
 		switch v.Segments()[1] {
-		case 13, 14:
-			return &RegistryList{
-				DockerLibraryRegistry: dockerLibraryRegistry,
-				E2eRegistry:           e2eRegistry,
-				EtcdRegistry:          etcdRegistry,
-				GcRegistry:            gcRegistry,
-				SampleRegistry:        sampleRegistry,
-			}, nil
-		case 15:
-			return &RegistryList{
-				DockerLibraryRegistry: dockerLibraryRegistry,
-				E2eRegistry:           e2eRegistry,
-				GcRegistry:            gcRegistry,
-				SampleRegistry:        sampleRegistry,
-			}, nil
-		case 16:
+		case 13, 14, 15, 16:
+			return nil, fmt.Errorf("version not supported for this build: %s", version)
+		case 17:
 			return &RegistryList{
-				DockerLibraryRegistry:   dockerLibraryRegistry,
 				E2eRegistry:             e2eRegistry,
+				DockerLibraryRegistry:   dockerLibraryRegistry,
 				GcRegistry:              gcRegistry,
 				GoogleContainerRegistry: googleContainerRegistry,
-				SampleRegistry:          sampleRegistry,
+				DockerGluster:           dockerGluster,
+				QuayIncubator:           quayIncubator,
 
-				// The following keys are used in the v1.16 registry list however their images
+				// The following keys are used in the v1.17 registry list however their images
 				// cannot be pulled as they are used as part of tests for checking image pull
 				// behavior. They are omitted from the resulting config.
 				// InvalidRegistry:         invalidRegistry,
 				// GcAuthenticatedRegistry: gcAuthenticatedRegistry,
+				// PrivateRegistry:         privateRegistry,
 			}, nil
-		case 17:
+		case 18:
 			return &RegistryList{
 				E2eRegistry:             e2eRegistry,
 				DockerLibraryRegistry:   dockerLibraryRegistry,
 				GcRegistry:              gcRegistry,
 				GoogleContainerRegistry: googleContainerRegistry,
 				DockerGluster:           dockerGluster,
 				QuayIncubator:           quayIncubator,
+				PromoterE2eRegistry:     promoterE2eRegistry,
 
-				// The following keys are used in the v1.17 registry list however their images
+				// The following keys are used in the v1.18 registry list however their images
 				// cannot be pulled as they are used as part of tests for checking image pull
 				// behavior. They are omitted from the resulting config.
 				// InvalidRegistry:         invalidRegistry,
 				// GcAuthenticatedRegistry: gcAuthenticatedRegistry,
 				// PrivateRegistry:         privateRegistry,
 			}, nil
-		case 18:
+		case 19:
 			return &RegistryList{
 				E2eRegistry:             e2eRegistry,
```

Comment:
This list also needs to include `BuildImageRegistry`, `E2eVolumeRegistry` and `SigStorageRegistry`.

Reply:
So, that is what I wasnt sure, how is that info determined?

Reply:
This function is used to create the YAML file when running `sonobuoy gen default-image-config` so it should include all the registries that form part of the [`v1_19` image list:](https://github.com/vmware-tanzu/sonobuoy/pull/1160/files/6f097931031ad12dc68dfa85445c4f040075d3bb#diff-4af7a4d28032f00415359ceaf5d98de4R18). This is just a helper so that if someone wants to provide their own custom repo config, they don't need to look in the source to figure out all the keys they need to override.

When you do `sonobuoy run --e2e-repo-config <config>` (using the config file generated from the command above), the values in that config file are then used to form the full registry path of where to push the image to. You can try this out using the `--dry-run` mode of the `images push` command.

```
sonobuoy gen default-image-config > config.yaml
# modify config.yaml to use your custom registry for all keys
sonobuoy images push --custom-registry my-registry --e2e-repo-config config.yaml --dry-run
```

The output from this should show your custom registry being used for all images.

In the `v1_19` function, every registry value used in a `Config` should be defined as the value from the `RegistryList`. This is _super_ awkward and confusing so please tell me if I've done a terrible job explaining it and I will try again!